cmd/collector/app/zipkin/zipkin_json.go

```diff
@@ -0,0 +1,215 @@
+	spans, err := decode(body)
+	if err != nil {
+		return nil, err
+	}
+
+	return spansToThrit(spans)
+}
+
+func decode(body []byte) ([]zipkinSpan, error) {
+	type zipkinSpans []zipkinSpan
+	var spans zipkinSpans
+
+	if err := json.Unmarshal(body, &spans); err != nil {
+		return nil, err
+	}
+
+	for _, span := range spans {
+		if len(span.TraceID) != 16 {
+			return nil, errTraceID
+		}
+		if len(span.ID) != 16 && len(span.ID) != 32 {
+			return nil, errSpanID
+		}
+		if len(span.ParentID) > 0 && len(span.ParentID) != 16 {
+			return nil, errParentID
+		}
+	}
+
+	return spans, nil
+}
+
+func spansToThrit(spans []zipkinSpan) ([]*zipkincore.Span, error) {
+	var tSpans []*zipkincore.Span
+	for _, span := range spans {
+		tSpan, err := spanToThrift(span)
+		if err != nil {
+			return nil, err
+		}
+		tSpans = append(tSpans, tSpan)
+	}
+
+	return tSpans, nil
+}
+
+func spanToThrift(span zipkinSpan) (*zipkincore.Span, error) {
+	id, err := strconv.ParseUint(span.ID, 16, 64)
+	if err != nil {
+		return nil, err
+	}
+	traceID, err := strconv.ParseUint(span.TraceID, 16, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	tSpan := &zipkincore.Span{
+		ID:        int64(id),
+		TraceID:   int64(traceID),
+		Name:      span.Name,
+		Debug:     span.Debug,
+		Timestamp: span.Timestamp,
+		Duration:  span.Duration,
+	}
+
+	if len(span.ParentID) > 0 {
+		parentID, err := strconv.ParseUint(span.ParentID, 16, 64)
+		if err != nil {
+			return nil, err
+		}
+		signed := int64(parentID)
+		tSpan.ParentID = &signed
+	}
+
+	for _, anno := range span.Annotations {
+		anno, err := annoToThrift(anno)
+		if err != nil {
+			return nil, err
+		}
+		tSpan.Annotations = append(tSpan.Annotations, anno)
+	}
+	for _, binAnno := range span.BinaryAnnotations {
+		binAnno, err := binAnnoToThrift(binAnno)
+		if err != nil {
+			return nil, err
+		}
+		tSpan.BinaryAnnotations = append(tSpan.BinaryAnnotations, binAnno)
+	}
+
+	return tSpan, nil
+}
+
+func endpointToThrift(endp endpoint) (*zipkincore.Endpoint, error) {
+	ipv4, err := parseIpv4(endp.IPv4)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zipkincore.Endpoint{
+		ServiceName: endp.ServiceName,
+		Port:        endp.Port,
+		Ipv4:        ipv4,
+	}, nil
+}
+
+func annoToThrift(anno annotation) (*zipkincore.Annotation, error) {
+	endpoint, err := endpointToThrift(anno.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zipkincore.Annotation{
+		Timestamp: anno.Timestamp,
+		Value:     anno.Value,
+		Host:      endpoint,
+	}, nil
+}
+
+func binAnnoToThrift(binAnno binaryAnnotation) (*zipkincore.BinaryAnnotation, error) {
+	endpoint, err := endpointToThrift(binAnno.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &zipkincore.BinaryAnnotation{
+		Key:            binAnno.Key,
+		Value:          []byte(binAnno.Value),
+		Host:           endpoint,
+		AnnotationType: zipkincore.AnnotationType_STRING,
+	}, nil
+}
+
+func parseIpv4(str string) (int32, error) {
```

Comment:
Yeah, we should sync it up with official IDL (should be backwards compatible) - #335